File: src/spec/resources/testprojects/phptest/tests/DefaultApiClientTest.php
<?php

namespace PetstoreClient\Tests;

use PHPUnit\Framework\TestCase;
use PetstoreClient\Configuration;
use PetstoreClient\DefaultApiClient;

class DefaultApiClientTest extends TestCase
{
    // -- TLS verification disabled --

    public function testMakesHttpsRequestWithVerifySslFalse(): void
    {
        $wiremockUrl = getenv('WIREMOCK_HTTPS_URL');

        $config = (new Configuration())
            ->setBaseUrl($wiremockUrl)
            ->setVerifySsl(false);

        $client = new DefaultApiClient($config);
        $response = $client->sendRequest('GET', $wiremockUrl . '/api/test', [], null);

        $this->assertSame(200, $response->statusCode);
        $this->assertStringContainsString('success', $response->body);
    }

    // -- Custom CA bundle --

    public function testMakesHttpsRequestWithCustomCaCert(): void
    {
        $wiremockUrl = getenv('WIREMOCK_HTTPS_URL');
        $caCertPath = getenv('CA_CERT_PATH');

        $config = (new Configuration())
            ->setBaseUrl($wiremockUrl)
            ->setVerifySsl(true)
            ->setSslCaCert($caCertPath);

        $client = new DefaultApiClient($config);
        $response = $client->sendRequest('GET', $wiremockUrl . '/api/test', [], null);

        $this->assertSame(200, $response->statusCode);
        $this->assertStringContainsString('success', $response->body);
    }

    // -- HTTP proxy --

    public function testMakesHttpRequestThroughProxy(): void
    {
        $wiremockUrl = getenv('WIREMOCK_HTTP_URL');
        $proxyUrl = getenv('PROXY_URL');

        $config = (new Configuration())
            ->setBaseUrl($wiremockUrl)
            ->setProxy($proxyUrl);

        $client = new DefaultApiClient($config);
        $response = $client->sendRequest('GET', $wiremockUrl . '/api/test', [], null);

        $this->assertSame(200, $response->statusCode);
        $this->assertStringContainsString('success', $response->body);
    }

    // -- HTTP proxy with TLS --

    public function testMakesHttpsRequestThroughProxyWithVerifySslFalse(): void
    {
        $wiremockUrl = getenv('WIREMOCK_HTTPS_URL');
        $proxyUrl = getenv('PROXY_URL');

        $config = (new Configuration())
            ->setBaseUrl($wiremockUrl)
            ->setProxy($proxyUrl)
            ->setVerifySsl(false);

        $client = new DefaultApiClient($config);
        $response = $client->sendRequest('GET', $wiremockUrl . '/api/test', [], null);

        $this->assertSame(200, $response->statusCode);
        $this->assertStringContainsString('success', $response->body);
    }
}

File: src/spec/resources/testprojects/csharptest/Tests/DefaultApiClientTest.cs
using PetstoreClient;
using Xunit;

namespace Tests;

[Collection("WireMockSquid")]
public class DefaultApiClientTest
{
    private readonly WireMockSquidFixture _fixture;

    public DefaultApiClientTest(WireMockSquidFixture fixture)
    {
        _fixture = fixture;
    }

    [Fact]
    public async Task MakesHttpsRequestWithVerifySslFalse()
    {
        var config = new Configuration
        {
            BaseUrl = _fixture.WireMockHttpsUrl,
            VerifySsl = false,
        };

        var client = new DefaultApiClient(config);
        var response = await client.SendRequestAsync(
            "GET",
            new Uri(_fixture.WireMockHttpsUrl + "/api/test"),
            new Dictionary<string, string>(),
            null
        );

        Assert.Equal(200, response.StatusCode);
        Assert.Contains("success", response.Body);
    }

    [Fact]
    public async Task MakesHttpsRequestWithCustomCaCert()
    {
        var config = new Configuration
        {
            BaseUrl = _fixture.WireMockHttpsUrl,
            VerifySsl = true,
            SslCaCert = _fixture.CaCertPath,
        };

        var client = new DefaultApiClient(config);
        var response = await client.SendRequestAsync(
            "GET",
            new Uri(_fixture.WireMockHttpsUrl + "/api/test"),
            new Dictionary<string, string>(),
            null
        );

        Assert.Equal(200, response.StatusCode);
        Assert.Contains("success", response.Body);
    }

    [Fact]
    public async Task MakesHttpRequestThroughProxy()
    {
        var config = new Configuration
        {
            BaseUrl = _fixture.WireMockHttpUrl,
            Proxy = _fixture.ProxyUrl,
        };

        var client = new DefaultApiClient(config);
        var response = await client.SendRequestAsync(
            "GET",
            new Uri(_fixture.WireMockHttpUrl + "/api/test"),
            new Dictionary<string, string>(),
            null
        );

        Assert.Equal(200, response.StatusCode);
        Assert.Contains("success", response.Body);
    }

    [Fact]
    public async Task MakesHttpsRequestThroughProxyWithVerifySslFalse()
    {
        var config = new Configuration
        {
            BaseUrl = _fixture.WireMockHttpsUrl,
            Proxy = _fixture.ProxyUrl,
            VerifySsl = false,
        };

        var client = new DefaultApiClient(config);
        var response = await client.SendRequestAsync(
            "GET",
            new Uri(_fixture.WireMockHttpsUrl + "/api/test"),
            new Dictionary<string, string>(),
            null
        );

        Assert.Equal(200, response.StatusCode);
        Assert.Contains("success", response.Body);
    }
}
